app/shared/environment.js

```diff
@@ -0,0 +1,53 @@
+/*
+ Copyright 2016 Mozilla
+
+ Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ this file except in compliance with the License. You may obtain a copy of the
+ License at http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software distributed
+ under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ specific language governing permissions and limitations under the License.
+ */
+/* global LIBDIR */
+
+import yargs from 'yargs';
+import path from 'path';
+import AppDirectory from 'appdirectory';
+import fs from 'fs-extra';
+import manifest from '../../package.json';
+import BUILD_CONFIG from '../../build-config.json';
+
+export const argParser = yargs.usage('Usage: $0 [options]').option('P', {
+  alias: 'profile',
+  default: undefined,
+  describe: 'The user profile directory.',
+  type: 'string',
+});
+
+let parsedArgs = undefined;
+
+export function parseArgs() {
+  if (parsedArgs !== undefined) {
```

Comment:
Let's document the decision and limitations here about only evaluating argv once